relay-metrics/src/aggregator/mod.rs

```diff
@@ -130,13 +131,21 @@ impl Aggregator {
     /// Pop and return the partitions with buckets that are eligible for flushing out according to
     /// bucket interval.
     ///
-    /// If no partitioning is enabled, the function will return a single `None` partition.
+    /// For each project `flush_decision` is called, which can influence the flush decision
+    /// for buckets of that project. The decision can also return metadata for the project
+    /// on which all flushed buckets for the project are merged with the `merge` function.
+    ///
+    /// `flush_decision` is only called once if the decision is [`FlushDecision::Flush`],
+    /// but may be called multiple times otherwise. The `flush_decision` should be consistent
+    /// for each project key passed.
     ///
-    /// Note that this function is primarily intended for tests.
-    pub fn pop_flush_buckets(
+    /// If no partitioning is enabled, the function will return a single `None` partition.
+    pub fn pop_flush_buckets<T>(
```

Comment:
```suggestion
    pub fn pop_flush_buckets<Buckets>(
```

Reply:
Don't think this is a good idea, because `Buckets` conflicts with the actual type `Buckets`.

Reply:
Oh I missed that that exists. I made the suggestion because it took me a while to understand that `T` represents a bucket container. Maybe `BucketContainer` or even `B`?

Alternative: `<T: BucketContainer>` where BucketContainer is a trait that implements `merge` (and remove the `merge` parameter from this function).